Stratis.Bitcoin.Features.Wallet/WalletSyncManager.cs

```diff
@@ -176,16 +191,26 @@ public virtual void ProcessBlock(Block block)
                         this.walletManager.ProcessBlock(nextblock, next);
                     }
                 }
```

Comment:
shouldn't we return here or not to process newTip block ? seems like it is processed twice then as we reach the end of the method, i think this is incorrect ...

Reply:
Actually we want to processes again, in case the puller was put back.

Reply:
i can again take your word on that, but are you sure in this case? like double-sure ? :-)

i mean, i was quite sure this is a bug, then you argued that double processing should be handled well, so i would change my classification from bug to non-optimal code, but you are saying here that you actually want to process it again, my mind is blown 

can you explain that? why would we want/need processing the same block again? i can't see it, tx